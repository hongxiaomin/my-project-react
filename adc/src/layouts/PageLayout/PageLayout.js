import React from 'react'
import { connect } from 'react-redux'
import PropTypes from 'prop-types'
import { setGlobalUrl } from 'helpers'
import { Dialog } from '@blueprintjs/core'
import LoginForm from 'containers/LoginForm'

export class PageLayout extends React.Component {
  static propTypes = {
    children: PropTypes.element.isRequired,
    lockScreen: PropTypes.bool.isRequired,
  }

  constructor() {
    super()
    this.state = {
      url: '172.22.34.220',
    }
  }

  componentDidMount() {
    setGlobalUrl(this.state.url)
  }

  render() {
    return (
      <div>
        {
          (this.props.lockScreen) ? (
            <Dialog
              isOpen
              style={{
                width: '540px',
              }}
            >
              <LoginForm />
            </Dialog>
          ) : null
        }
        {this.props.children}
      </div>
    )
  }
}

export default connect(state => ({
  lockScreen: state.auth.lockScreen,
}))(PageLayout)
