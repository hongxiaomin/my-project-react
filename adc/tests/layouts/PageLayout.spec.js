import React from 'react'
import { PageLayout } from 'layouts/PageLayout/PageLayout'
import { shallow, mount } from 'enzyme'

describe('(Layout) PageLayout', () => {
  it('renders its children inside of the viewport', () => {
    const Child = () => <h2>child</h2>
    const dispatch = jest.fn()
    const token = ''
    const lockScreen = false
    const pageLayoutWrapper = mount(
      <PageLayout dispatch={dispatch} token={token} lockScreen={lockScreen}>
        <Child />
      </PageLayout>,
    )
    // expect(pageLayoutWrapper.type()).toEqual('div')
    expect(pageLayoutWrapper).toMatchSnapshot()
  })
  it('renders LockScreen = true', () => {
    const Child = () => <h2>child</h2>
    const dispatch = jest.fn()
    const token = ''
    const lockScreen = true
    const pageLayoutWrapper = shallow(
      <PageLayout dispatch={dispatch} token={token} lockScreen={lockScreen}>
        <Child />
      </PageLayout>,
    )
    // expect(pageLayoutWrapper.type()).toEqual('div')
    expect(pageLayoutWrapper).toMatchSnapshot()
  })
})
