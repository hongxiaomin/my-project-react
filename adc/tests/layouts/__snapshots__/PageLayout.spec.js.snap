// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`(Layout) PageLayout renders LockScreen = true 1`] = `
<div>
  <Blueprint.Dialog
    isOpen={true}
    style={
      Object {
        "width": "540px",
      }
    }
  >
    <Connect(LoginForm) />
  </Blueprint.Dialog>
  <Child />
</div>
`;

exports[`(Layout) PageLayout renders its children inside of the viewport 1`] = `
<PageLayout
  dispatch={[Function]}
  lockScreen={false}
  token=""
>
  <div>
    <Child>
      <h2>
        child
      </h2>
    </Child>
  </div>
</PageLayout>
`;
